src/sageworks/artifacts/endpoints/endpoint.py

```diff
@@ -427,9 +427,26 @@ def capture_performance_metrics(
         # Compute the SHAP values
         shap_vals = self.shap_values(model_artifact, X_pred)
 
-        # Write shap vals to S3 Model Inference Folder
-        self.log.debug(f"Writing SHAP values to {self.model_inference_path}/inference_shap_values.csv")
-        wr.s3.to_csv(shap_vals, f"{self.model_inference_path}/inference_shap_values.csv", index=False)
+        # Multiple shap vals CSV for classifiers
+        if model_type == ModelType.CLASSIFIER.value:
+
+            # Need a separate shapley values CSV for each class
+            for i,class_shap_vals in enumerate(shap_vals):
+                df_shap = pd.DataFrame(class_shap_vals, columns=X_pred.columns)
+            
+                # Write shap vals to S3 Model Inference Folder
+                self.log.debug(f"Writing SHAP values to {self.model_inference_path}/inference_shap_values.csv")
+                wr.s3.to_csv(df_shap, f"{self.model_inference_path}/inference_shap_values_class_{i}.csv", index=False)
+
+        # Single shap vals CSV for regressors
+        if model_type == ModelType.REGRESSOR.value:
+
+            # Format shap values into single dataframe
+            df_shap = pd.DataFrame(shap_vals, columns=X_pred.columns)
+
+            # Write shap vals to S3 Model Inference Folder
+            self.log.debug(f"Writing SHAP values to {self.model_inference_path}/inference_shap_values.csv")
+            wr.s3.to_csv(df_shap, f"{self.model_inference_path}/inference_shap_values.csv", index=False)
 
```

Comment:
It's fine but both endpoint.py and model.py are getting to be a LOT of lines of code. We need to think about pulling out some logic when/where we can. Shrug... perhaps most of `capture_performance_metrics` can be split out into some utility methods. I'll take that as an action item...

Reply:
The issue I ran into with writing some of those methods as separate classes is that I ended up making a circular import from the Model class, or whatever artifact class I wanted to make methods for.

I generally don't feel like it's an issue to have a lot of lines of code in a single file when it pertains to a single class, but maybe there's a higher level reasoning to that.